lib/ts/recipe/passwordless/components/features/linkClickedScreen/index.tsx

```diff
@@ -0,0 +1,119 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Imports.
+ */
+/** @jsx jsx */
+import { jsx } from "@emotion/react";
+import { Fragment, PureComponent } from "react";
+
+import { FeatureBaseProps } from "../../../../../types";
+import { getQueryParams, getRedirectToPathFromURL, getURLHash } from "../../../../../utils";
+import FeatureWrapper from "../../../../../components/featureWrapper";
+import { StyleProvider } from "../../../../../styles/styleContext";
+import { defaultPalette } from "../../../../../styles/styles";
+import { getStyles } from "../../themes/styles";
+import {} from "../../../types";
+import { LinkClickedScreen as LinkClickedScreenTheme } from "../../themes/linkClickedScreen";
+import Recipe from "../../../recipe";
+import { ComponentOverrideContext } from "../../../../../components/componentOverride/componentOverrideContext";
+
+type PropType = FeatureBaseProps & { recipe: Recipe };
+
+class LinkClickedScreen extends PureComponent<PropType, unknown> {
+    getIsEmbedded = (): boolean => {
+        if (this.props.isEmbedded !== undefined) {
+            return this.props.isEmbedded;
+        }
+        return false;
+    };
+
+    componentDidMount = async (): Promise<void> => {
+        try {
+            const preAuthSessionId = getQueryParams("preAuthSessionId");
+            const linkCode = getURLHash();
+
+            if (preAuthSessionId === null || preAuthSessionId.length === 0 || linkCode.length === 0) {
+                return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                    error: "signin",
+                });
+            }
+
+            const response = await this.props.recipe.recipeImpl.consumeCode({
+                preAuthSessionId,
+                linkCode,
+                config: this.props.recipe.config,
+            });
+
+            if (response.status === "GENERAL_ERROR") {
+                return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                    error: "custom",
+                    message: response.message,
+                });
+            }
+
+            if (response.status === "RESTART_FLOW_ERROR") {
+                return this.props.recipe.redirectToAuthWithoutRedirectToPath(undefined, this.props.history, {
+                    error: "restart_link",
+                });
+            }
+            if (response.status === "OK") {
+                await this.props.recipe.recipeImpl.clearLoginAttemptInfo();
+                return this.props.recipe.redirect(
+                    { action: "SUCCESS", isNewUser: response.createdUser, redirectToPath: getRedirectToPathFromURL() },
```

Comment:
- The magic link won't have the custom redirect path. You need to save this in localstorage along with device info and fetch it from there.
- You should add a test for testing custom redirection during link flow and user input code flow